File: 진우/6weeks/2745. 진법 변환/진법 변환.cc
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;

int main() {
    string N;
    int B;
    cin >> N >> B;

    int len = N.size();
    reverse(N.begin(), N.end());
    
    int result = 0;
    int num = 0;

    for (int i = 0; i < len; i++) {
        if ('A' <= N[i] && N[i] <= 'Z') {
            num = (N[i] - 'A') + 10;
        }
        else {
            num = N[i] - '0';
        }

        int X = 1;
        for (int j = 0; j < i; j++) {
            X *= B;
        }

        result += num * X;
    }

    cout << result << "\n";

    return 0;
}

File: 진우/6weeks/2745. 진법 변환/진법 변환.py
N, B = input().split()
B = int(B)

result = 0
power = 1  # B^0

# 뒤에서부터 계산
for i in range(len(N) - 1, -1, -1):
    c = N[i]
    if 'A' <= c <= 'Z':
        num = ord(c) - ord('A') + 10
    else:
        num = ord(c) - ord('0')
    
    result += num * power
    power *= B

print(result)
